changelog.md

```diff
@@ -1,6 +1,15 @@
 This file contains a summary of changes to Haskell.nix and `nix-tools`
 that will impact users.
 
+## Feb 17, 2021
+* `ghcOptions` has been moved from package and is now a list of strings.
```

Comment:
Why the move? Presumably the idea is that at some point in the future cabal might give us some per-package ghc-options and then putting them on `package` would make sense, no? Or is the idea that they could come per-component? But I'm not sure cabal will let you set per-component options...

List of strings is great, though.

Reply:
The move is needed to make it work like the other "global options".  We want to support a `cabal.project` file with:

```
package *
  ghc-options: -g3
```

We can't currently convert that as it is not captured in the `plan.json` output (if it was it would be all good).  Since it is not we want to at least support adding a module:

```
modules = [{ ghcOptions = ["-g3"]; }];
```

We have code in pace to make this sort of "global option", but they do not end up in the `package` attribute.  Rather than making it a special case somehow it seems better just to move it.  It just happens that the "global options" also wind up at the component level.

The option to add them at the component level is not something `cabal.project` currently supports, however it seems likely that if they are put in the `plan.json` file they may be stored on the `component` node (so it might still be useful there).

Reply:
I meant moving where the option lives. Can we not do the same thing while still leaving the per-package option in `packages.foo.package.ghcOptions` rather than `packages.foo.ghcOptions`? My understanding was that the former represents "stuff we get from cabal"?

Reply:
> The option to add them at the component level is not something cabal.project currently supports, however it seems likely that if they are put in the plan.json file they may be stored on the component node (so it might still be useful there).

That's a good point, though. I guess we don't know where they'll end up so we're guessing regardless.

Reply:
As far as I can tell the options that can be set global are created [here](https://github.com/input-output-hk/haskell.nix/blob/d5563f3abee7c0e76ed56672b4bf8e94e1b98470/modules/plan.nix#L244) from `packageOptions`.  Those same `packageOptions` are exposed again [here for package level](https://github.com/input-output-hk/haskell.nix/blob/d5563f3abee7c0e76ed56672b4bf8e94e1b98470/modules/package.nix#L31) and [here for component level](https://github.com/input-output-hk/haskell.nix/blob/d5563f3abee7c0e76ed56672b4bf8e94e1b98470/modules/package.nix#L144).  The [package section](https://github.com/input-output-hk/haskell.nix/blob/d5563f3abee7c0e76ed56672b4bf8e94e1b98470/modules/package.nix#L36) does not seem to include support for setting the options global in this way.

Reply:
> My understanding was that the former represents "stuff we get from cabal"?

The only thing that sets `package.ghcOptions` is `stack-to-nix` from the `stack.yaml` file and user modules.  When `ghc-options` are set in a `.cabal` file they do not make it into `nix` at all.  Instead they are picked up by `setup configure` and used by `setup build` (cabal-to-nix does not do anything with `ghcOptions`).

In the future the `cabal.project` file and `plan.json` might provide GHC options.